LibDeflate.lua

```diff
@@ -2825,6 +2825,70 @@ do
 		, _fix_block_dist_huffman_bitlen, 31, 5)
 end
 
+
+--- Decompress a gzip compressed data.
+-- @param str [string] The data to be decompressed
+-- @return [string/nil] If the decompression succeeds, return the decompressed
+-- data. If the decompression fails, return nil. You should check if this return
+-- value is non-nil to know if the decompression succeeds.
+-- @return [integer] If the decompression succeeds, return the number of
+-- unprocessed bytes in the input compressed data. This return value is a
+-- positive integer if the input data is a valid compressed data appended by an
+-- arbitary non-empty string. This return value is 0 if the input data does not
+-- contain any extra bytes.<br>
+-- If the decompression fails (The first return value of this function is nil),
+-- this return value is undefined.
+function LibDeflate:DecompressGzip(str)
+	local arg_valid, arg_err = IsValidArguments(str)
+	if not arg_valid then
+		error(("Usage: LibDeflate:DecompressGzip(str): "..arg_err), 2)
+	end
+	if string_byte(string.sub(str, 1, 1)) ~= 31 or string_byte(string.sub(str, 2, 2)) ~= 139 then
+		error("LibDeflate:DecompressGzip(str): Not valid gzip data", 2)
```

Comment:
Should not throw when function argument type is valid, but data cant be decompressed.
return nil, error_code in this case.
For example,
return nil, 1